source/UIDataCodeGen/CodeGen/InstantiatorGeneratorBase.cs

```
@@ -3199,15 +3221,15 @@ sealed class ObjectData : Graph.Node<ObjectData>
 
             // The name that is given to the node by the NodeNamer. This name is used to generate factory method
             // names and field names.
-            public string Name { get; set; }
+            public string? Name { get; set; }
 
-            public string FieldName => RequiresStorage ? CamelCase(Name) : null;
+            public string? FieldName => RequiresStorage ? CamelCase(Name!) : null;
```

>Name! [](start = 68, length = 5)

Should this check if `Name` is null? If `Name` is null then `FieldName` would just be null anyway-- unless you explicitly want to raise an Exception here if `Name` is null.

---

I think this one can use one of the attributes. What do you think is the right thing to do here?


---
In reply to: [498966203](https://github.com/windows-toolkit/Lottie-Windows/pull/361#discussion_r498966203) [](ancestors = 498966203)

---

The code as written is indicating that we expect the Name value is non-null at this point. Should we add code to check that when CamelCase will blow up anyway if it's not true. There is a balance between making errors easy to find and the cost of extra code for checking errors. If the errors are going to be detected by some other mechanism (CamelCase blowing up in this case) then the extra code may not be worth it.


---
In reply to: [499018274](https://github.com/windows-toolkit/Lottie-Windows/pull/361#discussion_r499018274) [](ancestors = 499018274,498966203)

---

It depends on whether it's okay to have a null `FieldName` due to `Name` being null. There's already a case where it's okay for `FieldName` to be null (!RequiresStorage), but would there ever be a use case where I don't have a `Name` but still want to use the (null) `FieldName` anyway? If not, then I think it's fine to just let CamelCase crash.

---
In reply to: [499030990](https://github.com/windows-toolkit/Lottie-Windows/pull/361#discussion_r499030990) [](ancestors = 499030990,499018274,498966203)

---

Name should never be null if the object is actually going to be used - we do a pass over the ObjectData's to set their Names. But that happens after they've been constructed. As a result Name has to be nullable. 
This is a general problem of using stateful objects: if you can't set the state in the constructor, the reference-typed state of an object has to be nullable.
Nullable reference types enables the compiler to help you remove unnecessary null state, but some state, like this, is just part of the design.
More powerful languages (e.g. Rust) probably would enable ObjectData to be written in a way that it is stateless, but I don't know how to solve the problem in C#.


---
In reply to: [499092239](https://github.com/windows-toolkit/Lottie-Windows/pull/361#discussion_r499092239) [](ancestors = 499092239,499030990,499018274,498966203)